node/network/gossip-support/src/lib.rs

```diff
@@ -267,11 +268,13 @@ where
 					update_gossip_topology(
 						sender,
 						our_index,
-						session_info.discovery_keys,
+						session_info.discovery_keys.clone(),
 						relay_parent,
 						session_index,
 					)
 					.await?;
+
+					self.update_authority_ids(sender, session_info.discovery_keys).await;
```

Comment:
@ordian I moved the logic for emitting the event to `gossip-support`. Now for every new session, the `UpdatedAuthorityIds` event gets emitted. The only problem is that I am not sure how to check if the authority-ids are different from the ones from the previous session. Are we storing anywhere the authority-ids from the previous session?

Reply:
`gossip-support` maintains `connected_authorities` and `connected_authorities_by_peer_id` mappings